### PR TITLE
Fix NPE in MDCContextMap

### PR DESCRIPTION
**Pull request by SiegfriedTobias1**
PR for https://issues.apache.org/jira/browse/LOG4J2-2939
Accomodate for the fact that MDC.getCopyOfContextMap() may return null.

### CODE REVIEW

**Review by carterkozak — COMMENTED** [review]
Thanks, the change looks good. Would you mind adding a test that fails without the fix to prevent regressions?

**Comment by carterkozak** [comment]
Thanks! I'll handle the merge and cherry-pick to relevant release branches.

**Comment by SiegfriedTobias1** [comment]
Thanks for the review. I added a test, as requested.

File: log4j-to-slf4j/src/test/java/org/apache/logging/slf4j/LoggerTest.java [hunk 1]
```diff
@@ -165,5 +166,12 @@ public void mdc() {
         assertThat(list.strList, hasSize(2));
         assertTrue("Incorrect year", list.strList.get(0).startsWith("2010"));
     }
+
+    @Test
+    public void mdcNullBacked() {
+        assertNull("Setup wrong", MDC.getCopyOfContextMap());
+        assertTrue(ThreadContext.isEmpty());
+        assertFalse(ThreadContext.containsKey("something"));
+    }
```
Comment (garydgregory) [inline log4j-to-slf4j/src/test/java/org/apache/logging/slf4j/LoggerTest.java:175]: I would separate these into two tests, one for each API.

Comment (garydgregory) [reply log4j-to-slf4j/src/test/java/org/apache/logging/slf4j/LoggerTest.java:175]: An edge case worth testing IMO is getting a null key.

Comment (SiegfriedTobias1) [reply log4j-to-slf4j/src/test/java/org/apache/logging/slf4j/LoggerTest.java:180]: Thank you for the feedback! I split the tests and added some tests concerning null keys.
